### PR TITLE
Add application RW permissions to the Okta role.

### PR DESCRIPTION
**Pull request by mdwn**
Application read/write permissions have been added to the Okta role along with modifications to the cache and to the OktaAccessPoint to support application object management.

### CODE REVIEW

File: lib/auth/api.go [hunk 1]
```diff
@@ -1241,6 +1256,21 @@ func (w *OktaWrapper) DeleteOktaAssignment(ctx context.Context, name string) err
 	return w.NoCache.DeleteOktaAssignment(ctx, name)
 }
 
+// CreateApp creates a new application resource.
+func (w *OktaWrapper) CreateApp(ctx context.Context, app types.Application) error {
+	return w.NoCache.CreateApp(ctx, app)
+}
+
+// UpdateApp updates an existing application resource.
+func (w *OktaWrapper) UpdateApp(ctx context.Context, app types.Application) error {
```
Comment (espadolini) [inline lib/auth/api.go:1265]: Is there ever a situation where we actually need separate `Create` and `Update`, as opposed to just using some `Upsert` operation (that currently doesn't exist for `App`, for whatever reason)?

File: lib/authz/permissions.go [hunk 2]
```diff
@@ -803,9 +803,11 @@ func definitionForBuiltinRole(clusterName string, recConfig types.SessionRecordi
 			types.RoleSpecV6{
 				Allow: types.RoleConditions{
 					Namespaces: []string{types.Wildcard},
+					AppLabels:  types.Labels{types.Wildcard: []string{types.Wildcard}},
```
Comment (espadolini) [inline lib/authz/permissions.go:806]: Why does the Okta service need to _use_ apps?

File: lib/authz/permissions.go [hunk 3]
```diff
@@ -803,9 +803,11 @@ func definitionForBuiltinRole(clusterName string, recConfig types.SessionRecordi
...
 					Rules: []types.Rule{
 						types.NewRule(types.KindEvent, services.RW()),
 						types.NewRule(types.KindAccessRequest, services.RO()),
+						types.NewRule(types.KindApp, services.RW()),
```
Comment (espadolini) [inline lib/authz/permissions.go:810]: Make sure to document the implications of this in the setup guides.

File: lib/auth/api.go (hunk 1 above)
Comment (mdwn) [reply lib/auth/api.go:1265]: As part of the Okta service it's not needed, but I can't account for other use cases.

**Review by espadolini — APPROVED** [review]

**Review by hugoShaka — APPROVED** [review]

**Pull request merged**